cli/flox-rust-sdk/src/providers/upgrade_checks.rs

```diff
@@ -0,0 +1,301 @@
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::time::SystemTime;
+
+use fslock::LockFile;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use tracing::debug;
+
+use crate::models::environment::UpgradeResult;
+
+#[derive(Debug, Error)]
+pub enum UpgradeChecksError {
+    #[error("Failed to acquire lock")]
+    Lock(#[source] fslock::Error),
+    #[error("Failed to read upgrade information")]
+    Read(#[source] std::io::Error),
+    #[error("Failed to write upgrade information")]
+    Write(#[source] std::io::Error),
+    #[error("Failed to parse upgrade information")]
+    Deserialize(#[source] serde_json::Error),
+    #[error("Failed to serialize upgrade information")]
+    Serialize(#[source] serde_json::Error),
+}
+
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
+pub struct UpgradeInformation {
+    /// The last time the upgrade check was performed
+    pub last_checked: SystemTime,
```

Comment:
I think that was mentioned before, but we use a number of different time representations throughout.
are we ok with storing `SystemTime` ie. something like `{ "ts_secods": 12345, "ts_nanos": "987554" }` or should we converge to an iso time using either `chrono` or `time`?

Reply:
I would prefer consistency. I can't remember why we use both `chrono` and `time`. Seems like you're using `SystemTime` for exactly the same use case as what we're using `time` for here:
https://github.com/flox/flox/blob/8778414b043705d97898eaee0a427c51da859fb8/cli/flox/src/commands/mod.rs#L436-L440